### ISSUE TITLE
Macros inside loops

### ISSUE THREAD

**Issue by ymer** [TidierOrg/TidierData.jl#41]
I have some confusion with macros inside loops. 

This code works:

```
dfs = []

car_type = "Audi"
df = DataFrames.DataFrame(i=[1])
df = @mutate(df, car = !!car_type)
push!(dfs, df)

car_type = "Toyota"
df = DataFrames.DataFrame(i=[1])
df = @mutate(df, car = !!car_type)
push!(dfs, df)

df = vcat(df1, df2)
```
```
2×2 DataFrame
 Row │ i      car    
     │ Int64  String 
─────┼───────────────
   1 │     1  Audi
   2 │     1  Toyota
```

It also works if I make a loop, with standard Julia inside it:

```
dfs = []
for car_type in ["Toyota", "Audi"]
    df = DataFrames.DataFrame(i=[1])
    df[!, :car] = fill(car_type, 1)
    push!(dfs, df)
end
df = vcat(dfs...)
```
```

2×2 DataFrame
 Row │ i      car    
     │ Int64  String 
─────┼───────────────
   1 │     1  Audi
   2 │     1  Toyota

```
But when I use a TiderData macro inside a loop, it produces two Toyotas:

```
dfs = []
for car_type in ["Toyota", "Audi"]
    df = DataFrames.DataFrame(i=[1])
    df = @mutate(df, car = !!car_type)
    push!(dfs, df)
end
df = vcat(dfs...)
```

```
2×2 DataFrame
 Row │ i      car    
     │ Int64  String 
─────┼───────────────
   1 │     1  Toyota
   2 │     1  Toyota

```

Also if I make a nested loop, it works with standard Julia:

```
dfs = []
for car_type in ["Toyota", "Audi"]
    for color in ["Red", "Blue"]
        df = DataFrames.DataFrame(i=[1])
        df[!, :car] = fill(car_type, 1)
        df[!, :color] = fill(color, 1)
        push!(dfs, df)
    end
end
df = vcat(dfs...)
```
```
4×3 DataFrame
 Row │ i      car     color  
     │ Int64  String  String 
─────┼───────────────────────
   1 │     1  Toyota  Red
   2 │     1  Toyota  Blue
   3 │     1  Audi    Red
   4 │     1  Audi    Blue
```

But if I use a TidierData macro inside a nested loop, it gives an error:

```
dfs = []
for car_type in ["Toyota", "Audi"]
    for color in ["Red", "Blue"]
        df = DataFrames.DataFrame(i=[1])
        df = @mutate(df, car = !!car_type, color = !!color)
        push!(dfs, df)
    end
end
df = vcat(dfs...)
```

`ERROR: LoadError: UndefVarError: color not defined`

**Comment by kdpsingh** [comment]
This issue is explained in the documentation at the very end of the page here: https://tidierorg.github.io/TidierData.jl/latest/examples/generated/UserGuide/interpolation/

It's not that macros don't work inside loops. It's that `!!` interpolation doesn't currently work inside of loops. The documentation contains a workaround.

There's an open issue related to fixing this: https://github.com/TidierOrg/TidierData.jl/issues/16

I'll go ahead and close this issue because we are already tracking it. For now, I would use the workaround from the docs.

**Comment by kdpsingh** [comment]
Before closing, I'll post a version in the comments that will work so that you have a way to move forward.

**Comment by kdpsingh** [comment]
This issue is now fixed in #44. Interpolation no longer requires global variables. See updated documentation here: https://tidierorg.github.io/TidierData.jl/latest/examples/generated/UserGuide/interpolation/

Your original code should work now.

An updated version of the package is on its way to the registry.
